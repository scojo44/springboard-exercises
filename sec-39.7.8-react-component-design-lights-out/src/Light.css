.Light {
  height: 100px;
  width: 100px;
  background-color: darkgray;
  cursor: pointer;
}

.Light .lit {
  background-color: white;
}