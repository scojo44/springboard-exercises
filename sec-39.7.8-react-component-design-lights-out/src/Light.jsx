import React from "react";
import "./Light.css";

/** A single light on the board.
 *
 * This has no state --- just two props:
 *
 * - flipLightsAroundMe: a function received from the board which flips this
 *      light and the lights around it
 *
 * - isLit: boolean, is this Light lit?
 *
 * This handles clicks --- by calling flipLightsAroundMe
 *
 **/

function Light({ isLit, flipLightsAroundMe }) {
  const classes = `Light ${isLit ? "Light-lit" : ""}`;
  return <td className={classes} onClick={flipLightsAroundMe} />;
}

export default Light;
