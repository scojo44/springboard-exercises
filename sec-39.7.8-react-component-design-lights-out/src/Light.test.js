import { render, fireEvent } from "@testing-library/react";
import Light from "./Light";

it('renders without crashing', () => {
  render(
    <table>
      <tbody>
        <tr>
          <Light islit={true} flipLightsAroundMe={e => console.log("Clicked!")} />
        </tr>
      </tbody>
    </table>
  );
});

it('matches the snapshot', () => {
  const {asFragment} = render(
    <table>
      <tbody>
        <tr>
          <Light islit={true} flipLightsAroundMe={e => console.log("Clicked!")} />
        </tr>
      </tbody>
    </table>
  );
  expect(asFragment()).toMatchSnapshot();
});
