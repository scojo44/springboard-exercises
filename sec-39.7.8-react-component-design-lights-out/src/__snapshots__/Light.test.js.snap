// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<DocumentFragment>
  <table>
    <tbody>
      <tr>
        <td
          class="Light "
        />
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;
